### PR TITLE
Add tests for VM template source and VM empty state

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

**Comment by rawagner** [comment]
@gouyang please take a look

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/support/index.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  cy.exec(
+    `kubectl delete --ignore-not-found=${ignoreNotFound} -n ${resource.metadata.namespace} --cascade ${kind} ${resource.metadata.name}`,
+  );
+});
+
+Cypress.Commands.add('createResource', (resource) => {
+  cy.exec(`echo '${JSON.stringify(resource)}' | kubectl create -f -`);
+});
+
+Cypress.Commands.add('createDataVolume', (name: string, namespace: string) => {
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/support/index.ts:29]: Can we put the dv json in fixture and just call `createResource` to do it?

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { ProvisionSource } from '../../integration-tests/tests/utils/constants/enums/provisionSource';
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts:1]: I prefer copying the file into `integration-tests-cypress` directory, let's don't have dependencies on the old tests, wdyt?

File: frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-source.tsx [hunk 3]
```diff
@@ -89,6 +89,7 @@ const DeleteSourceButton: React.FC<DeleteSourceButtonProps> = ({ template, sourc
           isInline
           variant={ButtonVariant.link}
           onClick={() => createDeleteSourceModal({ sourceStatus })}
+          data-test="delete-template-source"
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-source.tsx:92]: Like this.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  [ProvisionSource.URL.getValue()]: 'input[id="provision-source-url"]',
+  [ProvisionSource.CONTAINER.getValue()]: 'input[id="provision-source-container"]',
+};
+
+type DiskSourceOpts = {
+  pvcName: string;
+  pvcNamespace: string;
+};
+
+export const addSource = {
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts:13]: It needs to take storageClass/accessMode/volueMode into considerations here, in the old tests we extract the values from configMap `KUBEVIRT_STORAGE_CLASS_DEFAULTS`, for simplicity I prefer to just use consts to define them for each storageClass. Either way works for me.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts (hunk 2 above)
Comment (rawagner) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts]: ok, I agree to not have deps on old tests. But I also dont like to just copy the file. I created a new directory `integration-tests-utils` where we can put pieces which are meant to be shared between new and old tests - and moved `ProvisionSource` there.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts (hunk 4 above)
Comment (rawagner) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts:13]: UI already sets the correct values. We dont need to change them in tests.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/support/index.ts (hunk 1 above)
Comment (rawagner) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/support/index.ts:27]: moved data volume to fixture. But I still need this command because I need to change the default name/namespace/access and volume modes. The command however uses `createResource`

File: frontend/packages/kubevirt-plugin/integration-tests-utils/enums/provisionSource.ts [hunk 5]
```diff
@@ -10,13 +10,15 @@ export class ProvisionSource extends ObjectEnum<string> {
   static readonly CONTAINER = new ProvisionSource(
     'Container',
     'Import via Registry (creates PVC)',
-    'quay.io/kubevirt/fedora-cloud-container-disk-demo:latest',
+    'quay.io/kubevirt/cirros-container-disk-demo:latest',
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/integration-tests-utils/enums/provisionSource.ts:13]: I changed the default container to cirros to make it smaller/faster

**Comment by rawagner** [comment]
@gouyang please take a look again

File: frontend/packages/kubevirt-plugin/integration-tests-utils/enums/provisionSource.ts (hunk 5 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests-utils/enums/provisionSource.ts:13]: We're using cirros long time ago and change it to fedora because some tests configured with cloud-init need to use fedora image, like guest-agent tests. 
How about let's have both of them and only use Fedora when necessary?

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts (hunk 4 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts:13]: Then let's add a step to change the storageClass from the environment variable.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/vm-template/source-image.spec.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    cy.login();
+    cy.visit('');
+    cy.createProject(testName);
+  });
+
+  beforeEach(() => {
+    virtualization.templates.visit();
+    cy.deleteResource({
+      kind: 'DataVolume',
+      metadata: { name: 'rhel6.9', namespace: 'kubevirt-os-images' },
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/vm-template/source-image.spec.ts:19]: `kubevirt-os-images` is `openshift-virtualization-os-images` in downstream, not sure what's the best way to address it, it at least can be a const on top of the tests, wdyt?

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts (hunk 4 above)
Comment (rawagner) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/view/add-source.ts:13]: Ok, you can specify `STORAGE_CLASS` env variable now.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/vm-template/source-image.spec.ts (hunk 6 above)
Comment (rawagner) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/vm-template/source-image.spec.ts]: added env variable `KUBEVIRT_OS_IMAGES_NS` which is set to `openshift-virtualization-os-images` by default

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/fixtures/data-volume.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  "kind": "DataVolume",
+  "metadata": {
+    "name": "foo",
+    "namespace": "bar",
+    "annotations": {
+      "cdi.kubevirt.io/storage.bind.immediate.requested": "true"
+    }
+  },
+  "spec": {
+    "pvc": {
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/fixtures/data-volume.json:12]: It needs to add storageClassName for the PVC to accept STORAG_CLASS from environment variable.

Comment (rawagner) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/fixtures/data-volume.json:12]: fixed

File: .gitignore [hunk 8]
```diff
@@ -19,6 +19,7 @@
 /frontend/packages/integration-tests-cypress/cypress-a11y-report.json
 /frontend/packages/dev-console/integration-tests/cypress-a11y-report.json
 /frontend/packages/ceph-storage-plugin/integration-tests-cypress/cypress-a11y-report.json
+/frontend/packages/kubevirt-plugin/integration-tests-cypress/cypress-a11y-report.json
```
Comment (spadgett) [inline .gitignore:22]: I don't think this is needed after #7916.

File: frontend/package.json [hunk 9]
```diff
@@ -37,6 +37,8 @@
     "test-cypress-devconsole-headless": "cd packages/dev-console/integration-tests && node --max-old-space-size=4096 ../../../node_modules/.bin/cypress run --env openshift=true --browser ${BRIDGE_E2E_BROWSER_NAME:=chrome} --headless --spec \"features/project-creation.feature\";",
     "test-cypress-ceph": "cd packages/ceph-storage-plugin/integration-tests-cypress && ../../../node_modules/.bin/cypress open --config-file cypress-ceph.json --env openshift=true",
     "test-cypress-ceph-headless": "cd packages/ceph-storage-plugin/integration-tests-cypress && node --max-old-space-size=4096 ../../../node_modules/.bin/cypress run --config-file cypress-ceph.json --env openshift=true --browser ${BRIDGE_E2E_BROWSER_NAME:=chrome} --headless",
+    "test-cypress-kubevirt": "cd packages/kubevirt-plugin/integration-tests-cypress && ../../../node_modules/.bin/cypress open --env openshift=true",
+    "test-cypress-kubevirt-headless": "cd packages/kubevirt-plugin/integration-tests-cypress && node --max-old-space-size=4096 ../../../node_modules/.bin/cypress run --env openshift=true --browser ${BRIDGE_E2E_BROWSER_NAME:=chrome} --headless",
```
Comment (spadgett) [inline frontend/package.json:41]: @dtaylor113 We should investigate a way to add these tests that avoids two need commands in the package.json file every time. Maybe a script that takes a parameter or env var.

Nothing to change in this PR, just an observation.

File: test-prow-e2e.sh [hunk 10]
```diff
@@ -49,4 +49,6 @@ elif [ "$SCENARIO" == "olmFull" ]; then
   ./test-cypress.sh -p olm -h true
 elif [ "$SCENARIO" == "ceph" ]; then
   ./test-cypress.sh -p ceph -h true
+elif [ "$SCENARIO" == "kubevirt-gating" ]; then
+  ./test-cypress.sh -p kubevirt -h true
```
Comment (spadgett) [inline test-prow-e2e.sh:53]: I assume you want to skip the Protractor tests as well (line 37 above).

File: frontend/package.json (hunk 9 above)
Comment (dtaylor113) [reply frontend/package.json:41]: Hi @spadgett, created https://issues.redhat.com/browse/CONSOLE-2538 to track

File: .gitignore (hunk 8 above)
Comment (rawagner) [reply .gitignore]: thats right, removed

File: test-prow-e2e.sh (hunk 10 above)
Comment (rawagner) [reply test-prow-e2e.sh:53]: No, we still want to execute proctator tests too. We will remove them once we rewrite protractor gating suite to cypress.

**Comment by spadgett** [comment]
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/plugins/index.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
+// This function is called when a project is opened or re-opened (e.g. due to
+// the project's config changing)
+const wp = require('@cypress/webpack-preprocessor');
+const fs = require('fs');
+
+module.exports = (on, config) => {
```
Comment (spadgett) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/plugins/index.js:6]: @dtaylor113 I assume all plugins are doing this. Is there a way we can keep the config in place instead of copying it across plugins?

**Review by spadgett — APPROVED** [review]
Hey @rawagner. Changes look good to me, but I will defer to @gouyang for final review.
